libres/lib/enkf/time_map.cpp

```diff
@@ -350,42 +359,51 @@ bool time_map_update(time_map_type *map, int step, time_t time) {
 }
 
 bool time_map_try_update(time_map_type *map, int step, time_t time) {
-    bool updateOK;
     time_map_assert_writable(map);
+    std::string error;
     pthread_rwlock_wrlock(&map->rw_lock);
-    { updateOK = time_map_update__(map, step, time); }
+    error = time_map_update__(map, step, time);
     pthread_rwlock_unlock(&map->rw_lock);
-    return updateOK;
+    return error.empty();
 }
 
-bool time_map_summary_update(time_map_type *map, const ecl_sum_type *ecl_sum) {
+/**
+Checks if there summary data and time map are in sync, returns
+an error string, empty string means the time map is in sync with
+the summary data.
+*/
+std::string time_map_summary_update(time_map_type *map,
+                                    const ecl_sum_type *ecl_sum) {
     time_map_assert_writable(map);
     bool updateOK = true;
     pthread_rwlock_wrlock(&map->rw_lock);
+    std::vector<std::string> errors;
     {
```

Comment:
Consider removing extra brackets.

Reply:
The reason I left it was to indicate where we needed the wrlock, however if it does not make it easier to ready I will be happy to remove it.

Reply:
I don't find it easier to read, but that's just me.